std/assembly/dataview.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  loadUnsafeWithOffset,
+} from "./internal/arraybuffer";
+
+export class DataView {
+  readonly buffer: ArrayBuffer;
+  readonly byteOffset: i32;
+  readonly byteLength: i32;
+
+  constructor (buffer: ArrayBuffer, byteOffset: i32, byteLength: i32) {
+    this.buffer = buffer
+    this.byteOffset = byteOffset
+    this.byteLength = byteLength
+  }
```

Comment:
You can improve constructor in this way: 
```ts
export class DataView { 
   constructor(
     public readonly buffer: ArrayBuffer,
     public readonly byteOffset: i32 = 0,
     public readonly byteLength: i32 = i32.MAX_VALUE
   ) {
     this.byteLength = min(this.byteLength, byteLength);
   }
}
```

Reply:
Neat!

Would you mind explaining this part?

```ts
this.byteLength = min(this.byteLength, byteLength);
```

It seems like `this.byteLength` and `byteLength` refers to the same variable in that case? and thus it shouldn't really do anything?

Or did you mean `buffer.byteLength`?

```ts
this.byteLength = min(buffer.byteLength, byteLength);
```

Reply:
Yeah, you are right, I mean `this.byteLength = min(buffer.byteLength, byteLength);`